### PR TITLE
Fix authentication with kerberized HDFS

### PR DESCRIPTION
**Pull request by bf8086**
#8236

Alluxio master currently initiates login for UFS by calling connectFromMaster when a UFS is mounted through commandline. However, this is not called when root ufs is mounted or when nested ufs is remounted on restart, therefore user has to rely on kinit to enable Alluxio to talk to Kerberized HDFS.

This change ensures before connecting to a UFS, it always logs in the user with the credentials from Alluxio properties, which frees up user from having to use kinit and activates HDFS client relogin mechanism, so user does not have to periodically kinit to refresh the Kerberos ticket.

### CODE REVIEW

**Comment by bf8086** [comment]
@gpang This is an improved fix for the Kerberized HDFS login issue. PTAL.

File: core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java [hunk 1]
```diff
@@ -127,10 +127,17 @@ private UnderFileSystem getOrAdd(AlluxioURI ufsUri, UnderFileSystemConfiguration
...
+      try {
+        connectUfs(fs);
+      } catch (IOException e) {
+        LOG.warn("Failed to perform initial connect to UFS {}: {}", ufsUri, e.getMessage());
+      }
       return fs;
     }
   }
 
+  protected abstract void connectUfs(UnderFileSystem fs) throws IOException;
```
Comment (gpang) [inline core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java]: can you add a javadoc on what this does?

**Comment by bf8086** [comment]
@gpang The ticket cache will be ignored in favor of the keytab login. Once the keytab is given, the server can talk to Kerberized HDFS even if the TGT from ticket cache expires.

**Comment by gpang** [comment]
alluxio-bot, merge this please

**Pull request merged**